src/components/webview/MainWebviewProvider.ts

```diff
@@ -115,160 +122,117 @@ export class IntuitaProvider implements WebviewViewProvider {
-					command: 'vscode.diff',
-					arguments: [null, element.job.oldContentUri, 'Delete File'],
-				};
-			}
+		mappedNode.children =
+			'children' in element
+				? element.children.map(this.__getTreeByCase)
+				: [];
 
-			if (element.job.kind === JobKind.moveAndRewriteFile) {
-				mappedNode.command = {
-					title: 'Move & Rewrite File',
-					command: 'vscode.diff',
-					arguments: [
-						element.job.oldContentUri,
-						element.job.newContentUri,
-						'Proposed change',
-					],
-				};
-			}
+		if (element.kind === ElementKind.CASE) {
+			mappedNode.children = element.children
+				.flatMap((fileElement) => fileElement.children)
+				.map(this.__getTreeByCase);
+		}
 
-			if (element.job.kind === JobKind.moveFile) {
-				mappedNode.command = {
-					title: 'Move File',
-					command: 'vscode.diff',
-					arguments: [
-						element.job.oldContentUri,
-						element.job.newContentUri,
-						'Proposed change',
-					],
-				};
-			}
+		return mappedNode;
+	};
 
-			mappedNode.actions = [
-				{
-					title: '✓ Apply',
-					command: 'intuita.acceptJob',
-					arguments: [element.job.hash],
-				},
-				{
-					title: '✗ Dismiss',
-					command: 'intuita.rejectJob',
-					arguments: [element.job.hash],
-				},
-			];
+	private __getTreeByDirectory = (element: Element): TreeNode | undefined => {
+		const mappedNode: TreeNode = {
+			id: element.hash,
+			iconName: getElementIconBaseName(element.kind),
+		};
 
-			if (this.__jobManager.isJobAccepted(element.jobHash)) {
-				mappedNode.kind = 'acceptedJobElement';
-
-				mappedNode.actions = [
-					{
-						title: '✗ Dismiss',
-						command: 'intuita.rejectCase',
-						arguments: [element.hash],
-					},
-					{
-						title: 'Issue',
-						command: 'intuita.createIssue',
-						arguments: [element.job.hash],
-					},
-					{
-						title: 'PR',
-						command: 'intuita.createPR',
-						arguments: [element.job.hash],
-					},
-				];
-			}
+		if (element.kind === ElementKind.JOB) {
+			this.__buildJobTree(element, mappedNode);
+			return {
+				...mappedNode,
+				label:
+					jobKindCopyMap[element.job.kind] +
+					` from ${element.job.codemodName}`,
+			};
 		}
 
-		if (element.kind === 'CASE') {
-			mappedNode.kind = 'caseElement';
-			mappedNode.actions = [
-				{
-					title: '✓ Apply',
-					command: 'intuita.acceptCase',
-					arguments: [element.hash],
-				},
-				{
-					title: '✗ Dismiss',
-					command: 'intuita.rejectCase',
-					arguments: [element.hash],
-				},
-			];
-
-			const caseJobHashes = this.__caseManager.getJobHashes([
-				String(element.hash) as CaseHash,
-			]);
-			const caseAccepted = Array.from(caseJobHashes).every((jobHash) =>
-				this.__jobManager.isJobAccepted(jobHash),
-			);
+		if (element.kind === ElementKind.ROOT) {
+			element.children.forEach(this.__getTreeByDirectory);
+			return {
+				...mappedNode,
+				label: 'Recipe',
+				children: this.__folderMap.has(ROOT_FOLDER_KEY)
+					? [this.__folderMap.get(ROOT_FOLDER_KEY) as TreeNode]
```

Comment:
ideally you could do this without casting with `as` by checking the return value of `.get` before the return statement

Reply:
I've encountered this problem. Do you know a workaround? https://stackoverflow.com/questions/70723319/object-is-possibly-undefined-using-es6-map-get-right-after-map-set